backend/Dockerfile

```diff
@@ -13,32 +26,40 @@ RUN apt update && apt install -y \
     gcc \
     && rm -rf /var/lib/apt/lists/*
 
-# install python libraries (except torch)
-COPY requirements/ requirements/
 ENV PIP_CERT=$CACERT_LOCATION
-RUN pip --default-timeout=300 install --upgrade pip \
-    && pip --default-timeout=300 install --no-cache-dir -r requirements/common.txt \
-    && rm -r /root/.cache
+
+# Download dependencies as a separate step to take advantage of Docker's caching.
+# Leverage a cache mount to /root/.cache/pip to speed up subsequent builds.
+# Leverage a bind mount to requirements.txt to avoid having to copy them into
+# into this layer.
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=bind,source=requirements.txt,target=requirements.txt \
+    python -m pip install -r requirements.txt

```

Comment:
- I personally need the default-timeout because otherwise it won't build on my work PC
- For installing pip packages it's always a good practice to first upgrade pip because otherwise it might not access the lastest versions of packages (kind of like apt-get update)
- why remove the rm /root/.cache which releases the cache installations from pip and makes the docker layer lighter ?

Reply:
Actually I just took everything from the default Dockerfile for Python, to stick to it as much as possible. Here the goal is to be able to rebuild the docker image quicker.

I made those changes only to follow the default Dockerfile provided by docker. If you prefer I can juste cancel them.